ui/pages/token-allowance/token-allowance.js

```diff
@@ -128,17 +145,36 @@ export default function TokenAllowance({
       fullTxData.originalApprovalAmount = dappProposedTokenAmount;
     }
 
```

Comment:
`fullTxData.txParams.data` is not updated anywhere, so if a user enters a custom amount, the approval will still be submitted with the default amount

Reply:
Good catch once again! :raised_hands:
I fixed it here: 679cab6b5c2fff30738a662f4f4365ff5b851286